python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -59,49 +128,128 @@ def initializeVariability(self, doCache=False):
+        # dict keyed on the names of the parameters required by
+        # the method method_name.  The values of this dict will
+        # be lists of parameter values for all astrophysical
+        # objects in the CatSim database.  Even objects that
+        # do no callon method_name will have entries in these
+        # lists (they will be set to None).
+        params = {}
+
+        for ix, varCmd in enumerate(varParams_arr):
+            if str(varCmd) != 'None':
+                varCmd = json.loads(varCmd)
+
+                # find the key associated with the name of
+                # the specific variability model to be applied
+                if 'varMethodName' in varCmd:
+                    meth_key = 'varMethodName'
+                else:
+                    meth_key = 'm'
+
+                # find the key associated with the list of
+                # parameters to be supplied to the variability
+                # model
+                if 'pars' in varCmd:
+                    par_key = 'pars'
+                else:
+                    par_key = 'p'
+            else:
+                # if there is no varParamStr, setup a null model
+                varCmd = {'varMethodName': 'None', 'pars':{}}
+                meth_key = 'varMethodName'
+                par_key = 'pars'
 
-        it uses then feeds the pars array to that method, under the assumption
-        that the parameters needed by the method can be found therein
+            # if we have discovered a new variability model
+            # that needs to be called, initialize its entries
+            # in the params dict
+            if varCmd[meth_key] not in method_name_arr:
+                params[varCmd[meth_key]] = {}
+                for p_name in varCmd[par_key]:
+                    params[varCmd[meth_key]][p_name] = [None]*len(varParams_arr)
 
-        @param [in] varParams is a string object (readable by json) that tells
-        us which variability model to use
+            method_name_arr.append(varCmd[meth_key])
+            for p_name in varCmd[par_key]:
+                params[varCmd[meth_key]][p_name][ix] = varCmd[par_key][p_name]
 
-        @param [out] output is a dict of magnitude offsets keyed to the filter name
-        e.g. output['u'] is the magnitude offset in the u band
+        method_name_arr = numpy.array(method_name_arr)
+        for method_name in params:
+            for p_name in params[method_name]:
+                params[method_name][p_name] = numpy.array(params[method_name][p_name])
 
-        """
-        if self.variabilityInitialized == False:
-            self.initializeVariability(doCache=True)
+        expmjd=None
 
-        varCmd = json.loads(varParams)
+        # Loop over all of the variability models that need to be called.
+        # Call each variability model on the astrophysical objects that
+        # require the model.  Add the result to deltaMag.
+        for method_name in numpy.unique(method_name_arr):
```

Comment:
No.  See above.  The `method_name_arr.append()` line occurs outside of the `if varCmd[meth_key] not in method_name_arr` block.